### PR TITLE
netwitness fetch incidents fix

### PR DESCRIPTION
**Pull request by glicht**
## Status
In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/15284

## Description
Raise exception when fetch incidents fails.

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Additional changes
Fixed regex of PACKAGE_YML_FILE_REGEX to include also the '.' character in package names.

### CODE REVIEW

**Comment by Itay4** [comment]
@kirbles19 please review Integrations/RSANetWitness_v11_1/CHANGELOG.md

File: Integrations/RSANetWitness_v11_1/RSANetWitness_v11_1.py [hunk 1]
```diff
@@ -941,27 +972,38 @@ def test_module():
...
+            delete_incident()
+        elif command == 'netwitness-get-alerts':
+            get_alerts()
+        sys.exit(0)
+    except ValueError as e:
+        LOG(e.message)
+        LOG.print_log()
+        if command == 'fetch-incidents':  # fetch-incidents supports only raising exceptions
+            raise
+        return_error_entry(e.message)
```
Comment (Itay4) [inline Integrations/RSANetWitness_v11_1/RSANetWitness_v11_1.py]: use `return_error`

File: Integrations/RSANetWitness_v11_1/RSANetWitness_v11_1.py [hunk 2]
```diff
@@ -941,27 +972,38 @@ def test_module():
...
+        sys.exit(1)
```
Comment (Itay4) [inline Integrations/RSANetWitness_v11_1/RSANetWitness_v11_1.py]: remove

File: Integrations/RSANetWitness_v11_1/RSANetWitness_v11_1.py [hunk 3]
```diff
@@ -941,27 +972,38 @@ def test_module():
...
+            get_incidents()
+        elif command == 'netwitness-update-incident':
+            update_incident()
+        elif command == 'netwitness-delete-incident':
+            delete_incident()
+        elif command == 'netwitness-get-alerts':
+            get_alerts()
+        sys.exit(0)
+    except ValueError as e:
+        LOG(e.message)
```
Comment (Itay4) [inline Integrations/RSANetWitness_v11_1/RSANetWitness_v11_1.py]: these should be done inside the condition

File: Integrations/RSANetWitness_v11_1/RSANetWitness_v11_1.py [hunk 4]
```diff
@@ -941,27 +972,38 @@ def test_module():
...
+            get_alerts()
+        elif command == 'netwitness-get-incidents':
+            get_incidents()
+        elif command == 'netwitness-update-incident':
+            update_incident()
+        elif command == 'netwitness-delete-incident':
+            delete_incident()
+        elif command == 'netwitness-get-alerts':
+            get_alerts()
+        sys.exit(0)
```
Comment (Itay4) [inline Integrations/RSANetWitness_v11_1/RSANetWitness_v11_1.py]: remove this

File: Integrations/RSANetWitness_v11_1/RSANetWitness_v11_1_test.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+def test_fetch_incidents(mocker):
+    def mock_demisto():
+        from RSANetWitness_v11_1 import get_timestamp
+        mocker.patch.object(demisto, "getLastRun", return_value={
+            "timestamp": get_timestamp("2019-08-13T09:56:02.000000")
+        })
+        mocker.patch.object(demisto, 'incidents')
+
+    from RSANetWitness_v11_1 import fetch_incidents
+    mock_demisto()
```
Comment (Itay4) [inline Integrations/RSANetWitness_v11_1/RSANetWitness_v11_1_test.py]: should verify results here

File: Integrations/RSANetWitness_v11_1/RSANetWitness_v11_1.py [hunk 6]
```diff
@@ -565,21 +566,50 @@ def get_alerts():
...
+        new_timestamp += 'Z'
+    timestamp_min_four_position = new_timestamp[-4]
+    if timestamp_min_four_position == ':':  # if contains no milisecs
+        new_timestamp = new_timestamp[:-1] + '.00000Z'
+    elif timestamp_min_four_position == '.':  # if contains only 3 milisecs
+        new_timestamp = new_timestamp[:-1] + '000Z'
+    try:
+        return datetime.strptime(new_timestamp, iso_format)
+    except ValueError:  # If wrong time format, will pass
+        pass
```
Comment (Itay4) [inline Integrations/RSANetWitness_v11_1/RSANetWitness_v11_1.py]: dont `pass` here, just raise `ValueError` here

**Comment by kirbles19** [comment]
@Itay4 Updated the changelog.md. Please check the timestamp RN for technical accuracy.

**Comment by glicht** [comment]
@yaakovi see the changes I regarding: Fixed regex of PACKAGE_YML_FILE_REGEX to include also the '.' character in package names.

**Comment by yaakovi** [comment]
build failed on `SymantecEndpointProtection_Test`.
we need to investigate why it fails with wrong IP (most of the times it works).

**Pull request merged**